test/src/unit-rest-array-schema-load.cc

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file unit-rest-array-schema-load.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests tiledb_array_schema_load* functions via a REST server.
+ */
+
+#include "test/support/src/vfs_helpers.h"
+#include "test/support/tdb_catch.h"
+#include "tiledb/api/c_api/enumeration/enumeration_api_internal.h"
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/c_api/tiledb_struct_def.h"
+#include "tiledb/sm/cpp_api/tiledb"
+#include "tiledb/sm/cpp_api/tiledb_experimental"
+
+using namespace tiledb;
+
+struct RESTArraySchemaLoadFx {
+  RESTArraySchemaLoadFx();
+  ~RESTArraySchemaLoadFx() = default;
+
+  void create_array();
+
+  test::VFSTestSetup vfs_test_setup_;
+  std::string uri_;
+  Context ctx_;
+};
+
+TEST_CASE_METHOD(
+    RESTArraySchemaLoadFx,
+    "Simple schema load test",
+    "[rest][array-schema][simple-load]") {
+  create_array();
+
+  ArraySchema schema = Array::load_schema(ctx_, uri_);
+  auto matcher = Catch::Matchers::ContainsSubstring(
```

Comment:
I think we should also check that the array schema and array schemas all are also correct.

Reply:
Done, this test here uses C++ API which uses post_array_schema_from_rest under the hood but the API only returns the latest schema. I added the checks for all array schemas in `unit-request-handlers.cc`.